### PR TITLE
Documentation restructuring & Policy docs

### PR DESCRIPTION
**Pull request by tgraf**

### CODE REVIEW

File: Documentation/concepts.rst [hunk 1]
```diff
@@ -431,24 +431,36 @@ for another local endpoint to the routing subsystem of the Linux kernel. This
 means that the packet will be routed as if a local process would have emitted
 the packet. As a result, the network connecting the cluster nodes must be aware
 that each of the node IP prefixes are reachable by using the node's primary IP
-address as an L3 next hop address. This is typically achieved using two
-methods:
+address as an L3 next hop address. 
+
+Cilium automatically enables IP forwarding in Linux when direct mode is
+configured, but it is up to the container cluster administrator to ensure that
+each routing element in the underlying network has a route that describe each
```
Comment (ianvernon) [inline Documentation/concepts.rst]: describes

File: Documentation/policy.rst [hunk 2]
```diff
@@ -0,0 +1,408 @@
...
+####################
+Policy Specification
+####################
+
+This chapter specifies the policy language used to configure network policies
+in Cilium. Security policies can be specified and imported via the following
+mechanisms:
+
+* The Kubernetes *NetworkPolicy* specification which offers to configure a
+  subset of the full Cilium security. For fun see `Kubernetes Network Policies
```
Comment (ianvernon) [inline Documentation/policy.rst]: 'a subset of the full Cilium security' is awkwardly phrased. What about "which configures a subset of what Cilium can secure"?

Comment (ianvernon) [inline Documentation/policy.rst]: For fun --> For more information

File: Documentation/policy.rst [hunk 3]
```diff
@@ -0,0 +1,408 @@
...
+
+*********
+Base Type
+*********
+
+All policy rules share a common base type which specifies what endpoints the
+rule applies to and also carries common metadata to identify the rule.
+
+Each rule is split into an ingress section which contains the rules which must
+be applied at ingress and egress of all endpoint matching the endpoint
```
Comment (ianvernon) [inline Documentation/policy.rst]: of all endpoints

File: Documentation/policy.rst [hunk 4]
```diff
@@ -0,0 +1,408 @@
...
+        }
+
+
+*****************
+L3 (Labels Based)
+*****************
+
+The L3 policy specifies which endpoints can talk to each other. L3 policies can
+be specified using Labels_ or CIDR. For CIDR, refer to the policy_cidr_
+section. Label based L3 policy can only be specified on ingress using the
```
Comment (ianvernon) [inline Documentation/policy.rst]: Label-based

File: Documentation/policy.rst [hunk 5]
```diff
@@ -0,0 +1,408 @@
...
+``fromEndpoints`` field.  ``fromEndpoints`` is a list of endpoints identified
+by a Kubernetes LabelSelector_ which are allowed to communicate with the
+endpoint subject to the rule.
+
+For more complex label combinations, the special ``fromRequires`` can be ued to
```
Comment (ianvernon) [inline Documentation/policy.rst]: used

File: Documentation/policy.rst [hunk 6]
```diff
@@ -0,0 +1,408 @@
...
+
+
+*****************
+L3 (Labels Based)
+*****************
+
+The L3 policy specifies which endpoints can talk to each other. L3 policies can
+be specified using Labels_ or CIDR. For CIDR, refer to the policy_cidr_
+section. Label based L3 policy can only be specified on ingress using the
+``fromEndpoints`` field.  ``fromEndpoints`` is a list of endpoints identified
```
Comment (ianvernon) [inline Documentation/policy.rst]: If used in tandem with Kubernetes, you can use ``fromEndpoints``, a list of endpoints...

File: Documentation/policy.rst [hunk 7]
```diff
@@ -0,0 +1,408 @@
...
+by a Kubernetes LabelSelector_ which are allowed to communicate with the
+endpoint subject to the rule.
+
+For more complex label combinations, the special ``fromRequires`` can be ued to
+establish label requirements which apply to multiple destinations.
+``fromRequires`` is a list of additional constraints which must be met in order
+for the selected endpoints to be reachable. These additional constraints do no
+by itself grant access privileges and must always be accompanied with at least
+one matching FromEndpoints.
```
Comment (ianvernon) [inline Documentation/policy.rst]: FromEndpoints -- ``fromEndpoints``

File: Documentation/policy.rst [hunk 8]
```diff
@@ -0,0 +1,408 @@
...
+
+        [{
+            "endpointSelector": {"matchLabels": {"role":"backend"}},
+            "ingress": [{
+                "fromEndpoints": [
+                  {"matchLabels":{"role":"frontend"}}
+                ]
+            }]
+        }]
+
```
Comment (ianvernon) [inline Documentation/policy.rst:162]: I think we should describe what this policy does similar to what we do with `Example (Requires)`

File: Documentation/policy.rst [hunk 9]
```diff
@@ -0,0 +1,408 @@
...
+Example (Requires)
+==================
+
+The following example builds on top of the previous one but requires that *all*
```
Comment (ianvernon) [inline Documentation/policy.rst]: "the previous one" --> link to the basic example within the doc

File: Documentation/policy.rst [hunk 10]
```diff
@@ -0,0 +1,408 @@
...
+        }
+
+
+.. _policy_l4:
+
+*********
+L4 Policy
+*********
+
+The L4 policy can be specified in addition to L3 policies. It restricts the
```
Comment (ianvernon) [inline Documentation/policy.rst]: The L4 Policy --> L4 policy

File: Documentation/policy.rst [hunk 11]
```diff
@@ -0,0 +1,408 @@
...
+            }]
+        }
+
+
+
+*********
+L7 (HTTP)
+*********
+
+Layer 7 policy can be specified embedeed into policy_l4_ rules. The ``L7Rules``
```
Comment (ianvernon) [inline Documentation/policy.rst]: embedeed --> embedded

File: Documentation/policy.rst [hunk 12]
```diff
@@ -0,0 +1,408 @@
...
+                        ]
+                    }
+                }]
+            }]
+        }
+
+
+
+.. _k8s_policy:
+
```
Comment (ianvernon) [inline Documentation/policy.rst]: General comment - we should have a way to be able to import all policies listed in these docs in a test to make sure the guides don't fall out of sync with master.

File: Documentation/policy.rst [hunk 13]
```diff
@@ -0,0 +1,408 @@
...
+                // Specs is a list of desired Cilium specific rule specification.
+                Specs api.Rules `json:"specs,omitempty"`
+        }
+
+A ``CiliumNetworkPolicy`` can carry either a single rule or a list of rules. If
+a list of rules are specified, all rules will be added and removed atomically.
+This is useful if multiple rules depend on each other and should not be applied
+independently.
+
+The ``Spec`` respecively ``Specs`` field refers to a standard `Cilium Policy Rule`_
```
Comment (ianvernon) [inline Documentation/policy.rst]: respectively

File: Documentation/troubleshooting.rst [hunk 14]
```diff
@@ -0,0 +1,203 @@
...
+Troubleshooting
+###############
+
+This document describes how to troubleshoot Cilium in different deployment
+modes. It focuses on a full deployment of Cilium within a datacenter or public
+cloud. If you are just looking for a simple way to experiment, we highly
+recommend trying out the :ref:`gs_guide` instead.
+
+This guide assumes that you have read the :ref:`arch_guide` which explains all
+the components and concepts.
```
Comment (ianvernon) [inline Documentation/troubleshooting.rst:13]: all of the components...

**Review by ianvernon — COMMENTED** [review]
Overall, looks great, just some minor typos / grammatical fixes.

File: Documentation/troubleshooting.rst [hunk 15]
```diff
@@ -0,0 +1,203 @@
...
+=============================
+
+IPv6 external connectivity can be tested with:
+
+.. code:: bash
+
+    ip -6 route get `host -t aaaa www.google.com | awk '{print $5}'`
+    ping6 www.google.com
+
+If the default route is missing, your VM may not be receiving router
```
Comment (danwent) [inline Documentation/troubleshooting.rst]: Calling this a VM is a bit odd.  Your "host"

File: Documentation/troubleshooting.rst [hunk 16]
```diff
@@ -0,0 +1,203 @@
...
+In this example, the cause for the failure is a Linux kernel running on the
+worker node which is not meeting :ref:`admin_system_reqs`.
+
+If the cause for the problem is not apparent based on these simple steps,
+please come and seek help on our `Slack channel`_.
+
+Testing External Connectivity
+=============================
+
+IPv6 external connectivity can be tested with:
```
Comment (danwent) [inline Documentation/troubleshooting.rst]: I'm curious why this jumps right to v6 connectivity testing, without showing IPv4.  Is the assumption that this is obvious, but that there are tricky issues with testing of v6?

File: Documentation/troubleshooting.rst [hunk 17]
```diff
@@ -0,0 +1,203 @@
...
+    PING www.google.com(zrh04s07-in-x04.1e100.net) 56 data bytes
+    64 bytes from zrh04s07-in-x04.1e100.net: icmp_seq=1 ttl=56 time=7.84 ms
+    64 bytes from zrh04s07-in-x04.1e100.net: icmp_seq=2 ttl=56 time=8.63 ms
+    64 bytes from zrh04s07-in-x04.1e100.net: icmp_seq=3 ttl=56 time=8.83 ms
+
+
+Monitoring Packet Drops
+=======================
+
+When connectivity is not as it should. A main cause an be unwanted packet drops
```
Comment (danwent) [inline Documentation/troubleshooting.rst]: "as it should be.  A main cause can be.."

File: Documentation/troubleshooting.rst (hunk 15 above)
Comment (tgraf) [reply Documentation/troubleshooting.rst]: All of this was existing documentation that I just moved to another file. I agree tough with the feedback and will fix it up

File: Documentation/troubleshooting.rst (hunk 16 above)
Comment (tgraf) [reply Documentation/troubleshooting.rst]: I'm removing this as this altogether as this is back from when direct routing mode was the default

**Comment by tgraf** [comment]
I think I've addressed all of the feedback.

**Pull request merged**